app/controllers/participants_controller.rb

```diff
@@ -1,9 +1,22 @@
 class ParticipantsController < ApplicationController
-  
+
   before_action :set_participant, only: [:show, :edit, :update, :destroy]
 
   # GET /participants
   # GET /participants.json
+
+
+
+    def assign_santas
+      secret_santas = Participant.all.shuffle.zip(Participant.all)
+      secret_santas.each do |santa_pair|
```

Comment:
Oh wow thank you. That makes it a lot simpler. Nice one. 
Any idea how I would make sure participants can’t get themselves as secret santas?
-- 
Owen Schebella
Sent with Airmail

On 14 March 2016 at 5:38:36 pm, MJ (notifications@github.com) wrote:

In app/controllers/participants_controller.rb:

>    before_action :set_participant, only: [:show, :edit, :update, :destroy]
>   
>    # GET /participants
>    # GET /participants.json
> +
> +
> +
> +    def assign_santas
> +      secret_santas = Participant.all.shuffle.zip(Participant.all)
> +      secret_santas.each do |santa_pair|
In ruby, you can iterate on tuples like so:

secret_santas.each do |giver, receiver|
  Assignment.create! giver: giver, receiver: receiver
end

—
Reply to this email directly or view it on GitHub.